component/telemetry.go

```diff
@@ -17,6 +17,10 @@ type TelemetrySettings struct {
 	// component to be used later as well.
 	Logger *zap.Logger
 
+	// SampledLogger passed to the created component.
+	// It will be used to avoid flooding the logs with messages that are repeated frequently.
+	SampledLogger *zap.Logger
```

Comment:
`SampledLogger` should always be built from `Logger`. Instead of doing it like this, if we want to have this as part of the public API, I think it is better to have a method on `TelemetrySettings` that generates the `SampledLogger` on first use (e.g. using `sync.Once` to initialize a private field).

Alternatively, I would question that we need this on `TelemetrySettings` (do we have any other use cases for it?). Can we do it so that this is internal to exporterhelper? Without changing the public API, the way to do that would be a private global variable initialized on first use.

Reply:
Great feedback. I will create the sampledLogger the first time using it